File: SDKs/netboxdemo.com/2.8/python/src/sdk/models/operations/dcim_power_panels_partial_update.py
from dataclasses import dataclass, field
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from sdk.models import shared


@dataclass
class DcimPowerPanelsPartialUpdatePathParams:
    id: int = field(metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    

@dataclass
class DcimPowerPanelsPartialUpdateRequest:
    path_params: DcimPowerPanelsPartialUpdatePathParams = field()
    request: shared.WritablePowerPanelInput = field(metadata={'request': { 'media_type': 'application/json' }})
    

@dataclass
class DcimPowerPanelsPartialUpdateResponse:
    content_type: str = field()
    status_code: int = field()
    power_panel: Optional[shared.PowerPanel] = field(default=None)
    

File: SDKs/netboxdemo.com/2.8/typescript/src/sdk/models/operations/dcimpowerpanelspartialupdate.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class DcimPowerPanelsPartialUpdatePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: number;
}


export class DcimPowerPanelsPartialUpdateRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: DcimPowerPanelsPartialUpdatePathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.WritablePowerPanelInput;
}


export class DcimPowerPanelsPartialUpdateResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  powerPanel?: shared.PowerPanel;

  @SpeakeasyMetadata()
  statusCode: number;
}
